astropy/_erfa/core.py.templ

```diff
@@ -118,46 +115,47 @@ def check_errwarn(statcodes, func_name):
         warnings.warn('ERFA function "' + func_name + '" yielded ' + wmsg, ErfaWarning)
 
 
-# <-------------------------trailing shape verification----------------------->
+# <------------------------structured dtype conversion------------------------>
 
-def check_trailing_shape(arr, shape, name):
-    try:
-        if arr.shape[-len(shape):] != shape:
-            raise Exception()
-    except:
-        raise ValueError("{0} must be of trailing dimensions {1}".format(name, shape))
+class D3Fix(numpy.ndarray):
+    # We do this to help ensure Quantity happily lets this pass...
+    def __quantity_subclass__(self, unit):
+        return type(self), True
+
+    def __array_wrap__(self, obj, context):
+        return obj.view(numpy.ndarray)
+
+
+def arrayify_inputs_and_create_d3_fix(inputs, core_dims, out_core_shape, out_dtype):
+    # Need to ensure we can get a shape
```

Comment:
This comment isn't very clear - could you write a comment to describe more verbosely what this function does?

Reply:
Now a proper docstring.